### ISSUE TITLE
Des decrypt raised 'Offset and length were out of bounds... '

### ISSUE THREAD

**Issue by TimRowe** [dotnet/corefx#24033]
My code like below. It works fine in .net framework4.6.1, But it raised exception in .net core 2.0.
 The exception raised in the line ` cs.Write(inputByteArray, 0, inputByteArray.Length);
` .
The message is `System.ArgumentException:“Offset and length were out of bounds for the array or count is greater than the number of elements from index to the end of the source collection.”
`.

Anyone help?

```c#
using System;
using System.Security.Cryptography;
using System.Text;

namespace ConsoleApp2
{
    class Program
    {
        public static string Md5Hash(string input)
        {
            MD5CryptoServiceProvider md5Hasher = new MD5CryptoServiceProvider();
            byte[] data = md5Hasher.ComputeHash(Encoding.Default.GetBytes(input));
            StringBuilder sBuilder = new StringBuilder();
            for (int i = 0; i < data.Length; i++)
            {
                sBuilder.Append(data[i].ToString("x2"));
            }
            return sBuilder.ToString().ToUpper();
        }


        public static string Decrypt(string Text, string sKey)
        {
            DESCryptoServiceProvider des = new DESCryptoServiceProvider();
            int len = 0;
            len = Text.Length / 2;
            byte[] inputByteArray = new byte[len];
            int x = 0;
            int i = 0;
            for (x = 0; x <= len - 1; x++)
            {
                i = Convert.ToInt32(Text.Substring(x * 2, 2), 16);
                inputByteArray[x] = Convert.ToByte(i);
            }
            des.Key = ASCIIEncoding.ASCII.GetBytes(Md5Hash(sKey).Substring(0, 8));
            des.IV = ASCIIEncoding.ASCII.GetBytes(Md5Hash(sKey).Substring(0, 8));
            System.IO.MemoryStream ms = new System.IO.MemoryStream();
            CryptoStream cs = new CryptoStream(ms, des.CreateDecryptor(), CryptoStreamMode.Write);
            cs.Write(inputByteArray, 0, inputByteArray.Length);
            cs.FlushFinalBlock();
            return Encoding.Default.GetString(ms.ToArray());
        }


        static void Main(string[] args)
        {
            var s = Decrypt("973335E697BFF06DC9F9481008526545458CCEB78D8717561617C2414464DBF451E44D4071EB03788CAA6B4560BA127493EF142A95AEF87349F4ADFDA82C5810FA2319F45E275057700474332D438849F7F26B63063EE6A3BD9D15354922C8E5965D20F075060C23FEFDF8C013A74D8640EE59D9DA687BD30CFD41B861F4DCF0294A4A4AF699738C8E6387275B742CE647EE733B572A8EDEAA099E854D11095D", "buyongji");
            Console.WriteLine(s);
        }
    }
}
```

**Comment by karelz** [comment]
@bartonjs can you check for compat concerns?

**Comment by bartonjs** [comment]
Exact dupe of https://github.com/dotnet/corefx/issues/23311.

**Comment by bartonjs** [comment]
https://github.com/dotnet/corefx/issues/23311#issuecomment-325385891 contains the workarounds (the easiest being to use `DES.Create()` instead of `new DESCryptoServiceProvider()`)

**Comment by karelz** [comment]
And as the dupe says, it will be fixed in next 2.0 servicing release.

**Comment by TimRowe** [comment]
@karelz  @bartonjs , Thanks for your help,  I used `DES.Create()`instead, it works well now.
